specs/node_types.md

```diff
@@ -0,0 +1,105 @@
+
+### Block Headers
+
+#### Compact Block Headers
+
+Nodes that only process compact block headers will download and validate the [block header](./data_structures.md#header) _without_ downloading or validating the `availableDataHeader` [block](./data_structures.md#block) field that is committed to in the block header. These nodes cannot perform Data Availability Sampling on [block bodies](#block-bodies).
+
+Secure under an honest supermajority of validator voting power and a weak subjectivity assumption.
+
+#### Extended Block Headers
+
+Nodes that process extended block headers will download and validate both the [compact block header](#compact-block-headers) and the 
+[`availabledataheader`](./data_structures.md##availabledataheader) [block](./data_structures.md#block) field. These nodes can perform Data Availability Sampling on [block bodies](#block-bodies), and their security assumptions depend on how block bodies are handled.
+
+### Block Bodies
+
+Block bodies (the `availableData` [block](./data_structures.md#block) field) can be downloaded and optionally stored and/or served. Storing and serving block body data has no effect on node security assumptions.
+
+#### No Bodies
+
+Nodes that only process [compact block headers](#compact-block-headers) have no need for block bodies and simply do not process block bodies.
+
+Secure under an honest supermajority majority of validator voting power and a weak subjectivity assumption.
+
+#### Sampled Bodies
+
+These nodes perform Data Availability Sampling on block bodies.
+
+Secure under an honest minority of nodes and a weak subjectivity assumption.
+
+#### Partial Bodies
+
+These nodes fully download and validate [the erasure coding](./data_structures.md#2d-reed-solomon-encoding-scheme) of a random subset of block bodies (configurable locally). Since the erasure coding of each block is stateless, nodes that perform validation of partial bodies contribute to the overall security of the network by being able to produce [fraud proofs of invalid erasure coding](./data_structures.md#invalid-erasure-coding).
+
+Secure under an honest minority of nodes and a weak subjectivity assumption.
+
+#### Full Bodies
+
+These nodes fully download and validate [the erasure coding](./data_structures.md#2d-reed-solomon-encoding-scheme) of all block bodies.
+
+If [transactions are not processed](#no-transactions), secure under an honest minority of nodes and a weak subjectivity assumption. If [transactions are processed](#full-transactions), secure under a weak subjectivity assumption.
+
+### Transactions
+
+#### No Transactions
+
+These nodes process do not process requests [with a reserved namespace ID](./data_structures.md#arranging-available-data-into-shares) and thus to not know the chain state without relying on a third party.
```

Comment:
Didn't we want to use transactions in favor of "requests"? #101

Reply:
I'd rather subsume all changes to the word "request" into a single PR and for now follow along with the same expression consistently.

Reply:
OK